assets/js/components/TourTooltips.js

```diff
@@ -0,0 +1,147 @@
+	disableAnimation: true,
+	styles: {
+		arrow: {
+			length: 8,
+			margin: 56,
+			spread: 16,
+		},
+		floater: {
+			filter: 'drop-shadow(rgba(60, 64, 67, 0.3) 0px 1px 2px) drop-shadow(rgba(60, 64, 67, 0.15) 0px 2px 6px)',
+		},
+	},
+};
+
+export default function TourTooltips( { steps, tourID } ) {
+	const stepKey = `${ tourID }-step`;
+	const runKey = `${ tourID }-run`;
+	const { setValue } = useDispatch( CORE_UI );
+	const [ shouldSkipTour, setShouldSkipTour ] = useLocalStorage( `sitekit-${ tourID }__has-encountered-tour` );
+
+	const stepIndex = useSelect( ( select ) => select( CORE_UI ).getValue( stepKey ) );
+	const run = useSelect( ( select ) => select( CORE_UI ).getValue( runKey ) || false );
+
+	const changeStep = ( index, action ) => setValue(
+		stepKey,
+		index + ( action === ACTIONS.PREV ? -1 : 1 ),
+	);
+
+	const startTour = () => {
+		// Checks if user has completed or skipped tour already, do not re-run tour if so.
+		if ( ! shouldSkipTour ) {
+			setValue( runKey, true );
+		}
+	};
+
+	const endTour = () => {
+		// Add to localStorage to avoid unwanted repeat viewing.
+		setShouldSkipTour( true );
+		setValue( runKey, false );
+	};
+
+	/**
+	 * Handles `react-joyride` state changes using callback function.
+	 *
+	 * @typedef {Object} JoyrideCallbackData
+	 * @property {string} action The action that updated the state.
+	 * @property {status} status The tour's status.
+	 * @property {number} index  Step index.
+	 * @property {string} type   Specific type (tour, step, beacon).
+	 *
+	 * @since n.e.x.t
+	 * @see {@link https://docs.react-joyride.com/callback} Example data provided by `react-joyride`.
+	 * @see {@link https://docs.react-joyride.com/constants} State & lifecycle constants used by `react-joyride`.
+	 *
+	 * @param {JoyrideCallbackData} data Data object provided via `react-joyride` callback prop.
+	 */
+	const handleJoyrideCallback = ( data ) => {
+		const { action, index, status, type } = data;
+
+		const hasCloseAction = action === ACTIONS.CLOSE;
+		const shouldChangeStep = ! hasCloseAction && [ EVENTS.STEP_AFTER, EVENTS.TARGET_NOT_FOUND ].includes( type );
+		const isFinishedOrSkipped = [ STATUS.FINISHED, STATUS.SKIPPED ].includes( status );
+		const shouldCloseFromButtonClick = hasCloseAction && type === EVENTS.STEP_AFTER;
+		const shouldEndTour = isFinishedOrSkipped || shouldCloseFromButtonClick;
+
+		if ( shouldChangeStep ) {
+			changeStep( index, action );
+		} else if ( shouldEndTour ) {
+			endTour();
+		}
+	};
+
+	// Start tour on initial render
+	useMount( startTour );
+
+	const parsedSteps = steps.map( ( step ) => ( {
+		disableBeacon: true,
+		isFixed: true,
+		placement: 'auto',
+		...step,
+	} ) );
+
+	return (
+		<Joyride
+			callback={ handleJoyrideCallback }
+			continuous
+			disableOverlayClose
+			floaterProps={ floaterProps }
+			run={ run }
+			showProgress
+			stepIndex={ stepIndex }
+			steps={ parsedSteps }
+			styles={ joyrideStyles }
+			tooltipComponent={ TourTooltip }
```

Comment:
We should provide all standard strings here via the `locale` prop, see https://docs.react-joyride.com/props. The individual `TourTooltip` component shouldn't need to override anything then and could use the provided props, making for a cleaner solution.